src/Shared/MSBuildLoadContext.cs

```diff
@@ -29,7 +29,7 @@ internal class MSBuildLoadContext : AssemblyLoadContext
                 "Microsoft.Build.Utilities.Core",
             }.ToImmutableHashSet();
 
-        internal static readonly string[] Extensions = new[] { "ni.dll", "ni.exe", "dll", "exe" };
+        internal static readonly string[] Extensions = new[] { "dll", "exe" };
```

Comment:
Nit: The .exe suffix does not really exist in .NET Core world either. Apps are managed .dlls in .NET Core too.

Reply:
Hmm, I believe that for managed exes we do generate the wrapper exe files running dll under the system-wide dotnet host, doesn't msbuild need to take them into account? Just curious, I'm not really familiar with msbuild source code, I just took a few quick peeks in the past when I was double-checking some details.

Reply:
The wrapper exe files are not managed assemblies. These extensions are only used for probe for managed assemblies only (it is obvious from the local context in which this static is used).

Reply:
I see, thanks for explaining!

Reply:
Thank you, not an urgent fix so I'll leave it up to @rainersigwald to address this when he's back.

Reply:
Yeah, I think removing it entirely is a good idea, so I did.